api/tests/integration/infrastructure/repositories/campaign-participation-overview-repository_test.js

```diff
@@ -14,209 +13,173 @@ describe('Integration | Repository | Campaign Participation Overview', () => {
 
-        databaseBuilder.factory.buildCampaignParticipationElementsForOverview({
-          userId,
-          index: '3',
-          isShared: true,
-          lastAssessmentState: Assessment.states.COMPLETED,
-          campaignParticipationCreatedAt: new Date('2000-07-03T10:00:00Z'),
-          campaignParticipationSharedAt: new Date('2000-07-03T10:00:00Z'),
-        });
+      context('the filter is ENDED', () => {
+        it('returns shared participation or completed participation with an archived campaign', async () => {
+          const states = ['ENDED'];
+          const { campaignParticipationOverviews } = await campaignParticipationOverviewRepository.findByUserIdWithFilters({ userId, states });
 
-        databaseBuilder.factory.buildCampaignParticipationElementsForOverview({
-          userId,
-          index: '4',
-          isShared: false,
-          lastAssessmentState: Assessment.states.STARTED,
-          campaignParticipationCreatedAt: new Date('2000-07-04T10:00:00Z'),
-          campaignArchivedAt: new Date('2000-07-05T11:00:00Z'),
+          expect(campaignParticipationOverviews[0].id).to.equal(endedParticipation.id);
+          expect(campaignParticipationOverviews).to.have.lengthOf(1);
         });
-
-        await databaseBuilder.commit();
       });
 
-      it('should retrieve all campaign participations of the user', async () => {
-        const rawCampaignParticipationOverviews = await campaignParticipationOverviewRepository.findByUserIdWithFilters({ userId });
-        const { campaignParticipationOverviews } = rawCampaignParticipationOverviews;
+      context('the filter is ARCHIVED', () => {
+        it('returns participation where the campaign is archived', async () => {
+          const states = ['ARCHIVED'];
+          const { campaignParticipationOverviews } = await campaignParticipationOverviewRepository.findByUserIdWithFilters({ userId, states });
 
-        expect(campaignParticipationOverviews).to.have.lengthOf(3);
-        expect(_.map(campaignParticipationOverviews, 'campaignTitle')).to.deep.equal(['2 - My campaign', '1 - My campaign', '3 - My campaign']);
+          expect(campaignParticipationOverviews[0].id).to.equal(archivedParticipation.id);
+          expect(campaignParticipationOverviews).to.have.lengthOf(1);
+        });
       });
 
-      it('should retrieve the campaign participations of the user with filtered assessment where state is TO_SHARE', async () => {
-        const states = ['TO_SHARE'];
-        const rawCampaignParticipationOverviews = await campaignParticipationOverviewRepository.findByUserIdWithFilters({ userId, states });
-        const { campaignParticipationOverviews } = rawCampaignParticipationOverviews;
-
-        expect(_.map(campaignParticipationOverviews, 'campaignTitle')).to.deep.equal(['2 - My campaign']);
+      context('the filters are ONGOING and TO_SHARE', () => {
+        it('returns shared participation or participation with a started assessment', async () => {
```

Comment:
Je pense que le nom du test n'est pas bon.
`'returns shared participation or participation with a started assessment'`
Car pour les filtres `ONGOING` et `TO_SHARE` on devrait avoir:
`'returns not shared participation with a completed assessment or participation with a started assessment'`

Reply:
Hum, je pense que parler d'`assessment` `completed` ou `started` dans ce test c'est pas une bonne idée c'est des détails trop technique.
tu pense que quoi de 
```js
context('when there are several statuses given for the status filter', () => {
  it('returns only participations which matches with the given statuses');
});

```

Reply:
Parfait  👍